module/zfs/dmu_send.c

```diff
@@ -2870,37 +2921,60 @@ dmu_adjust_send_estimate_for_indirects(dsl_dataset_t *ds, uint64_t uncompressed,
 }
 
 int
-dmu_send_estimate_fast(dsl_dataset_t *ds, dsl_dataset_t *fromds,
+dmu_send_estimate_fast(dsl_dataset_t *origds, dsl_dataset_t *fromds,
     zfs_bookmark_phys_t *frombook, boolean_t stream_compressed, uint64_t *sizep)
 {
 	int err;
+	dsl_dataset_t *ds = origds;
 	uint64_t uncomp, comp;
+	char dsname[ZFS_MAX_DATASET_NAME_LEN + 6];
 
-	ASSERT(dsl_pool_config_held(ds->ds_dir->dd_pool));
+	ASSERT(dsl_pool_config_held(origds->ds_dir->dd_pool));
 	ASSERT(fromds == NULL || frombook == NULL);
 
+	/*
+	 * If this is a saved send we may actually be sending
+	 * from the %recv clone used for resuming.
+	 */
+	if (!ds->ds_is_snapshot) {
+		dsl_dataset_name(origds, dsname);
+		(void) strcat(dsname, "/%recv");
```

Comment:
same constant `"%recv"` here.  Also, I think `dsname` can be declared inside the `if`.